plugins/org.jkiss.dbeaver.model.lsm/src/org/jkiss/dbeaver/model/lsm/LSMAnalyzer.java

```diff
@@ -16,19 +16,18 @@
  */
 package org.jkiss.dbeaver.model.lsm;
 
+import org.antlr.v4.runtime.ANTLRErrorListener;
 import org.jkiss.code.NotNull;
 import org.jkiss.code.Nullable;
-import org.jkiss.utils.Pair;
+import org.jkiss.dbeaver.model.stm.STMSource;
+import org.jkiss.dbeaver.model.stm.STMTreeRuleNode;
 
-import java.util.List;
-import java.util.concurrent.Future;
-
-
-public interface LSMAnalysis<T extends LSMElement> {    
+public interface LSMAnalyzer {
 
     @Nullable
-    Future<T> getModel();
+    STMTreeRuleNode parseSqlQueryTree(@NotNull STMSource source, @Nullable ANTLRErrorListener errorListener);
```

Comment:
Let's use some wrapper around ANTLRErrorListener to avoid direct imports of ANTLR packages everywhere.